xbmc/addons/AddonManager.h

```diff
@@ -244,9 +245,27 @@ namespace ADDON
 
     bool IsSystemAddon(const std::string& id);
 
-    bool AddToUpdateBlacklist(const std::string& id);
-    bool RemoveFromUpdateBlacklist(const std::string& id);
-    bool IsBlacklisted(const std::string& id) const;
+    /* \brief Add a single update rule to the list for an addon
+     * \sa CAddonUpdateRules::AddRuleToList()
+     */
+    bool AddRuleToList(const std::string& id, AddonUpdateRule updateRule);
+
+    /* \brief Remove all rules from update rules list for an addon
+     * \sa CAddonUpdateRules::RemoveAllRulesFromList()
+     */
+    bool RemoveAllRulesFromList(const std::string& id);
+
+    /* \brief Remove a specific rule from update rules list for an addon
+     * \sa CAddonUpdateRules::RemoveRuleFromList()
+     */
+    bool RemoveRuleFromList(const std::string& id, AddonUpdateRule updateRule);
+
+    /* \brief Check if an addon version is auto-updateable
+     * \param id addon id to be checked
+     * \return true is addon is auto-updateable, false otherwise
+     * \sa CAddonUpdateRules::IsAutoUpdateable()
+     */
+    bool IsAutoUpdateable(const std::string& id) const;
```

Comment:
@AlwinEsch i added the group but left the refs out because i think this may change over time and could probably be forgotten. if you insist on them i'll put 'em in though